src/main/java/it/pagopa/pn/deliverypush/service/utils/WebhookUtils.java

```diff
@@ -106,6 +114,33 @@ public EventEntity buildEventEntity(Long atomicCounterUpdated, StreamEntity stre
         return eventEntity;
     }
 
+    public TimelineElementInternal getTimelineInternalFromEvent(EventEntity entity) {
+        TimelineElementEntity timelineElementEntity = this.timelineElementJsonConverter.jsonToEntity(entity.getElement());
+        return entityToDtoTimelineMapper.entityToDto(timelineElementEntity);
+    }
+
+    public Mono<StreamEntity> verifyVersion(List<String> xPagopaPnCxGroups, String xPagopaPnApiVersion, StreamEntity stream) {
+        var isStreamVersionNull = org.apache.commons.lang3.StringUtils.isBlank(stream.getVersion());
```

Comment:
credo si possa semplificare, sfruttando il metodo getVersion fatto da voi.
in pratica:
```
int intHeaderVersion = getVersion(xPAgopaApiversion)
int intStreamVersion =  getVersion(stream.version) 
if ( intHeaderVersion == && intStreamVersion  == 10)
{
return stream;
}
if (intHeaderVersion  >= intStreamVersion 
&& intStreamVersion  > 10 
&& (CollectionUtils.isEmpty(xPagopaPnCxGroups) || WebhookUtils.checkGroups(stream.getGroups(), xPagopaPnCxGroups)))
{
return stream;
}
return Mono.error(new PnWebhookForbiddenException("Not supported operation"));
```

Reply:
inoltre, nella classe base WebhookServiceImpl c'era già un metodo getStreamEntityToRead.
Non è che quello fa al caso vostro per recuperare lo stream in lettura (già smazzandosi i problemi di visibilità/accesso)?

Reply:
abbiamo analizzato il getStreamEntityToRead,è all'interno del metodo viene fatta la query su streamEntityDao che viene già effettuata a monte nel metodo consumeEventStream

Reply:
si esatto, l'idea è che quel metodo recupera lo streamentity, facendo già i controlli che son comuni a più metodi, tra cui il consumeEvent (quindi se te lo ritorna, lo puoi usare, sennò lancia exception)

Reply:
modifica effettuata